mautic/mautic
# Company based marketing

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | n
| New feature? | y 
| Related user documentation PR URL | https://github.com/mautic/documentation/pull/99
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | 
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
This PR creates a new company so contacts can now belong to a company, and allows segments to be filtered by company.

#### Steps to test this PR:
1. Apply this PR run migrations
2. test that the new company menu is there and that everything works
3. on the list of contacts you can now batch assign leads to a company.
4. on a contact's detail view you can now see the companies it belongs to.
5. on a contact's detail view you can add or remove a company
6. on the company's list view you can see the number of leads in each company
7. on the list of contacts you can now filter by company_name:nameOfCompany
8. create some companies, add contacts to a company, then create a segment with filters that 9. would apply to your values.
10. you should now have contacts in your segment that meet both contact filters and company filters criteria.
11. create a new campaign and select action to add a new company, test campaign
12. try to submit a form with all required fields to identify a company, and find the lead being assigned to that company.

[//]: # ( As applicable: )
#### Steps to reproduce the bug:
1. 
2. 

#### List deprecations along with the new alternative:
1. 
2. 

#### List backwards compatibility breaks:
1. 
2.
